CI: TestWatchedServerConfigRotation received unexpected `[]tls.Certificate`

## CI failure

<details>
  <summary>Click to expand test output</summary>

```
--- FAIL: TestWatchedServerConfigRotation (0.40s)

    server_test.go:166: 
        	Error Trace:	/home/vagrant/go/src/github.com/cilium/cilium/pkg/crypto/certloader/server_test.go:166
        	Error:      	Not equal: 
        	            	expected: []tls.Certificate{tls.Certificate{Certificate:[][]uint8{[]uint8{0x30, 0x82, 0x3, 0x61, 0x30, 0x82, 0x2, 0x49, 0xa0, 0x3, 0x2, 0x1, 0x2, 0x2, 0x11, 0x0, 0xa5, 0x30, 0x7f, 0x7f, 0x9a, 0x6b, 0x2, 0x11, 0xe1, 0xab, 0x8a, 0xd8, 0x21, 0xea, 0xe5, 0xbb, 0x30, 0xd, 0x6, 0x9, 0x2a, 0x86, 0x48, 0x86, 0xf7, 0xd, 0x1, 0x1, 0xb, 0x5, 0x0, 0x30, 0x1e, 0x31, 0x1c, 0x30, 0x1a, 0x6, 0x3, 0x55, 0x4, 0x3, 0x13, 0x13, 0x68, 0x75, 0x62, 0x62, 0x6c, 0x65, 0x2d, 0x63, 0x61, 0x2e, 0x63, 0x69, 0x6c, 0x69, 0x75, 0x6d, 0x2e, 0x69, 0x6f, 0x30, 0x1e, 0x17, 0xd, 0x32, 0x30, 0x31, 0x30, 0x30, 0x32, 0x31, 0x34, 0x33, 0x34, 0x32, 0x39, 0x5a, 0x17, 0xd, 0x32, 0x33, 0x31, 0x30, 0x30, 0x32, 0x31, 0x34, 0x33, 0x34, 0x32, 0x39, 0x5a, 0x30, 0x2a, 0x31, 0x28, 0x30, 0x26, 0x6, 0x3, 0x55, 0x4, 0x3, 0xc, 0x1f, 0x2a, 0x2e, 0x64, 0x65, 0x66, 0x61, 0x75, 0x6c, 0x74, 0x2e, 0x68, 0x75, 0x62, 0x62, 0x6c, 0x65, 0x2d, 0x67, 0x72, 0x70, 0x63, 0x2e, 0x63, 0x69, 0x6c, 0x69, 0x75, 0x6d, 0x2e, 0x69, 0x6f, 0x30, 0x82, 0x1, 0x22, 0x30, 0xd, 0x6, 0x9, 0x2a, 0x86, 0x48, 0x86, 0xf7, 0xd, 0x1, 0x1, 0x1, 0x5, 0x0, 0x3, 0x82, 0x1, 0xf, 0x0, 0x30, 0x82, 0x1, 0xa, 0x2, 0x82, 0x1, 0x1, 0x0, 0xcd, 0x27, 0x8e, 0x63, 0xb1, 0x25, 0x2e, 0x15, 0xf3, 0xb0, 0xe0, 0xa6, 0x3f, 0x46, 0x9d, 0x5a, 0xdb, 0x85, 0xff, 0xe8, 0x16, 0xc3, 0x4e, 0x36, 0x4e, 0x62, 0x1d, 0xe2, 0x97, 0x79, 0x2, 0x7d, 0x45, 0xe7, 0xfa, 0x46, 0xcf, 0xe5, 0x1b, 0xc6, 0x9, 0x5f, 0x61, 0x8, 0x93, 0x6f, 0x7b, 0x7c, 0xe8, 0x79, 0x5, 0xe2, 0xce, 0xe8, 0x6, 0x33, 0xb5, 0x2f, 0xb0, 0x19, 0xd0, 0xa3, 0x45, 0xe, 0x6b, 0xc5, 0x1a, 0xee, 0xe2, 0x91, 0xfc, 0xf, 0xc5, 0x57, 0x9, 0x63, 0x17, 0xb0, 0x3c, 0x80, 0xa3, 0x2b, 0x90, 0x55, 0x11, 0x1e, 0x2f, 0x93, 0x35, 0x48, 0xef, 0xc3, 0xeb, 0xc2, 0x97, 0xe6, 0xa6, 0x67, 0x22, 0xa7, 0x92, 0xce, 0xb9, 0xd8, 0x43, 0xac, 0x8, 0x22, 0x20, 0x41, 0xfc, 0xab, 0x5f, 0xbf, 0xf, 0xcd, 0x85, 0xd2, 0xfb, 0x27, 0xcd, 0xc2, 0x13, 0x8a, 0x8f, 0x3e, 0x11, 0xdd, 0xb1, 0xfc, 0x89, 0x4e, 0xd3, 0x5b, 0xb, 0x63, 0xdd, 0x29, 0x3f, 0x32, 0xd7, 0x16, 0xc3, 0x5d, 0xae, 0xf2, 0xc5, 0x29, 0xb, 0xb1, 0xeb, 0xc4, 0x2b, 0x3d, 0xf4, 0xa7, 0xa8, 0x24, 0xdd, 0xd, 0x41, 0xa4, 0x28, 0xd, 0xf7, 0xe1, 0x8f, 0x6d, 0x70, 0x3c, 0x2b, 0xca, 0x52, 0xf1, 0x29, 0x42, 0x90, 0xdf, 0x7b, 0x2e, 0x96, 0x4d, 0x2f, 0xb, 0x5e, 0xbb, 0x31, 0xdd, 0x14, 0x80, 0x93, 0xcb, 0x6b, 0x76, 0xe5, 0x67, 0x64, 0x37, 0x6f, 0x3c, 0x91, 0x15, 0x93, 0x99, 0x67, 0xec, 0x13, 0xb3, 0xc2, 0x22, 0x2e, 0xa6, 0x36, 0xd1, 0xe4, 0x36, 0x3c, 0xd7, 0x63, 0x1a, 0x8f, 0x73, 0x33, 0x9d, 0xd5, 0x11, 0xb, 0x58, 0xb, 0xdd, 0x62, 0x2b, 0x95, 0xe7, 0xce, 0xf8, 0x57, 0xff, 0x36, 0xf3, 0x3e, 0x9a, 0xc8, 0xe, 0x10, 0xaf, 0x6a, 0xfc, 0x20, 0xa6, 0x13, 0xea, 0xdc, 0xf2, 0x1a, 0xd1, 0x2, 0x3, 0x1, 0x0, 0x1, 0xa3, 0x81, 0x8d, 0x30, 0x81, 0x8a, 0x30, 0xe, 0x6, 0x3, 0x55, 0x1d, 0xf, 0x1, 0x1, 0xff, 0x4, 0x4, 0x3, 0x2, 0x5, 0xa0, 0x30, 0x1d, 0x6, 0x3, 0x55, 0x1d, 0x25, 0x4, 0x16, 0x30, 0x14, 0x6, 0x8, 0x2b, 0x6, 0x1, 0x5, 0x5, 0x7, 0x3, 0x1, 0x6, 0x8, 0x2b, 0x6, 0x1, 0x5, 0x5, 0x7, 0x3, 0x2, 0x30, 0xc, 0x6, 0x3, 0x55, 0x1d, 0x13, 0x1, 0x1, 0xff, 0x4, 0x2, 0x30, 0x0, 0x30, 0x1f, 0x6, 0x3, 0x55, 0x1d, 0x23, 0x4, 0x18, 0x30, 0x16, 0x80, 0x14, 0x4e, 0x89, 0xea, 0xff, 0xc, 0x9d, 0x2a, 0x66, 0x70, 0x40, 0x38, 0x23, 0x8f, 0x8a, 0x86, 0x36, 0x6e, 0xb8, 0x9a, 0xc1, 0x30, 0x2a, 0x6, 0x3, 0x55, 0x1d, 0x11, 0x4, 0x23, 0x30, 0x21, 0x82, 0x1f, 0x2a, 0x2e, 0x64, 0x65, 0x66, 0x61, 0x75, 0x6c, 0x74, 0x2e, 0x68, 0x75, 0x62, 0x62, 0x6c, 0x65, 0x2d, 0x67, 0x72, 0x70, 0x63, 0x2e, 0x63, 0x69, 0x6c, 0x69, 0x75, 0x6d, 0x2e, 0x69, 0x6f, 0x30, 0xd, 0x6, 0x9, 0x2a, 0x86, 0x48, 0x86, 0xf7, 0xd, 0x1, 0x1, 0xb, 0x5, 0x0, 0x3, 0x82, 0x1, 0x1, 0x0, 0xa7, 0x22, 0x11, 0x7c, 0xe0, 0x5e, 0x8b, 0x54, 0x9a, 0x71, 0xfd, 0x8c, 0x4a, 0x61, 0xf0, 0xb6, 0xed, 0xec, 0xae, 0xf, 0xca, 0x6b, 0xb7, 0xa6, 0xf0, 0x53, 0xef, 0x65, 0xad, 0x17, 0xe2, 0x94, 0x2e, 0xf3, 0xb9, 0x37, 0xc5, 0x9d, 0x2e, 0xcb, 0x9d, 0x24, 0x79, 0x70, 0x4e, 0x19, 0xae, 0xe8, 0xaf, 0x1e, 0x1, 0xd0, 0xb5, 0x70, 0x9d, 0x27, 0x87, 0xff, 0xe4, 0x4c, 0xc7, 0xd, 0xce, 0x9a, 0xba, 0xea, 0x2c, 0x98, 0x6b, 0xf2, 0x3b, 0x12, 0x7c, 0x4d, 0x31, 0xb6, 0x29, 0x91, 0x12, 0x17, 0xe5, 0xd0, 0x99, 0x6e, 0xbc, 0x58, 0x29, 0x68, 0x2d, 0xc6, 0x53, 0xe9, 0xc3, 0x2f, 0x8c, 0x13, 0xd3, 0x58, 0x29, 0x5e, 0xb2, 0xb4, 0x5f, 0x83, 0xc9, 0x81, 0x41, 0x58, 0xc5, 0x1a, 0xd7, 0xf1, 0x3e, 0xea, 0xa2, 0xd6, 0xb4, 0x2e, 0x7a, 0xb6, 0x94, 0x3f, 0xb4, 0xf2, 0x6e, 0xd4, 0x2d, 0x69, 0x2d, 0xf4, 0x52, 0x61, 0x21, 0x1f, 0x56, 0xf9, 0x8d, 0x86, 0xb1, 0x96, 0xb2, 0x57, 0xcf, 0x97, 0xd2, 0x28, 0x54, 0x17, 0x6, 0x47, 0xdb, 0x8e, 0x2, 0x8, 0x8f, 0xa3, 0x72, 0xb9, 0x84, 0x11, 0x2e, 0x83, 0x88, 0xfa, 0xb1, 0xb1, 0xa, 0xd5, 0x6f, 0x54, 0xd5, 0x47, 0xe2, 0x81, 0x48, 0x1, 0x41, 0x14, 0x7e, 0x70, 0x58, 0x99, 0x34, 0xd5, 0xbf, 0xff, 0x1d, 0xe, 0x93, 0x4c, 0x73, 0x73, 0xb1, 0x14, 0x43, 0xdd, 0xf8, 0x2c, 0x65, 0x87, 0xba, 0xf7, 0x90, 0x1, 0x7c, 0x49, 0xc9, 0xd3, 0x56, 0x2f, 0xb3, 0x75, 0x5a, 0x64, 0xfc, 0x79, 0x78, 0x0, 0x45, 0xbb, 0x15, 0xc6, 0x46, 0x14, 0x11, 0xd5, 0xa, 0xfe, 0x82, 0x17, 0x36, 0x6e, 0x2e, 0xb2, 0x36, 0xe0, 0xf8, 0x1a, 0xb1, 0x5a, 0xf6, 0x30, 0x52, 0x2, 0x9d, 0x54, 0x78, 0xa9, 0x9f, 0x57, 0xc0, 0x93, 0x57, 0xbf, 0x9d, 0x96}}, PrivateKey:(*rsa.PrivateKey)(0xc0000b1020), SupportedSignatureAlgorithms:[]tls.SignatureScheme(nil), OCSPStaple:[]uint8(nil), SignedCertificateTimestamps:[][]uint8(nil), Leaf:(*x509.Certificate)(nil)}}
        	            	actual  : []tls.Certificate{tls.Certificate{Certificate:[][]uint8{[]uint8{0x30, 0x82, 0x3, 0x60, 0x30, 0x82, 0x2, 0x48, 0xa0, 0x3, 0x2, 0x1, 0x2, 0x2, 0x10, 0x71, 0x0, 0xb4, 0xe2, 0x8c, 0x4f, 0x26, 0x5e, 0xdc, 0xa2, 0xd5, 0x5a, 0x49, 0xec, 0xe5, 0xcf, 0x30, 0xd, 0x6, 0x9, 0x2a, 0x86, 0x48, 0x86, 0xf7, 0xd, 0x1, 0x1, 0xb, 0x5, 0x0, 0x30, 0x1e, 0x31, 0x1c, 0x30, 0x1a, 0x6, 0x3, 0x55, 0x4, 0x3, 0x13, 0x13, 0x68, 0x75, 0x62, 0x62, 0x6c, 0x65, 0x2d, 0x63, 0x61, 0x2e, 0x63, 0x69, 0x6c, 0x69, 0x75, 0x6d, 0x2e, 0x69, 0x6f, 0x30, 0x1e, 0x17, 0xd, 0x32, 0x30, 0x31, 0x30, 0x30, 0x31, 0x31, 0x33, 0x32, 0x35, 0x33, 0x31, 0x5a, 0x17, 0xd, 0x32, 0x33, 0x31, 0x30, 0x30, 0x31, 0x31, 0x33, 0x32, 0x35, 0x33, 0x31, 0x5a, 0x30, 0x2a, 0x31, 0x28, 0x30, 0x26, 0x6, 0x3, 0x55, 0x4, 0x3, 0xc, 0x1f, 0x2a, 0x2e, 0x64, 0x65, 0x66, 0x61, 0x75, 0x6c, 0x74, 0x2e, 0x68, 0x75, 0x62, 0x62, 0x6c, 0x65, 0x2d, 0x67, 0x72, 0x70, 0x63, 0x2e, 0x63, 0x69, 0x6c, 0x69, 0x75, 0x6d, 0x2e, 0x69, 0x6f, 0x30, 0x82, 0x1, 0x22, 0x30, 0xd, 0x6, 0x9, 0x2a, 0x86, 0x48, 0x86, 0xf7, 0xd, 0x1, 0x1, 0x1, 0x5, 0x0, 0x3, 0x82, 0x1, 0xf, 0x0, 0x30, 0x82, 0x1, 0xa, 0x2, 0x82, 0x1, 0x1, 0x0, 0xda, 0xf, 0xc9, 0xd, 0x9f, 0x31, 0xb9, 0x5c, 0x71, 0x64, 0x6, 0x79, 0x7f, 0xf3, 0x9b, 0x77, 0x2c, 0x7d, 0x6e, 0x42, 0x27, 0xdd, 0x32, 0x62, 0xd2, 0x73, 0x95, 0xf2, 0x6a, 0x24, 0x62, 0xb3, 0x30, 0x4c, 0x6a, 0xc1, 0xc, 0xee, 0x51, 0x25, 0x21, 0xaa, 0xf1, 0x9d, 0x3f, 0x6, 0x5d, 0xb7, 0x1a, 0xc0, 0x0, 0x39, 0x11, 0x3a, 0x20, 0xd8, 0xc6, 0x9c, 0xb4, 0x7d, 0x17, 0x16, 0x1a, 0xb0, 0x3e, 0x6d, 0xf1, 0xae, 0xf7, 0x9, 0xab, 0x94, 0x18, 0x8a, 0x84, 0x9d, 0x1e, 0x6f, 0x27, 0x2f, 0x3b, 0xac, 0x34, 0xce, 0xeb, 0x2e, 0xfe, 0xbc, 0x30, 0xa3, 0xf7, 0x1a, 0xe0, 0x44, 0x38, 0xcd, 0x61, 0x70, 0x7, 0xb3, 0x89, 0x4d, 0x71, 0x1a, 0x87, 0x30, 0x23, 0xc2, 0x6d, 0xf1, 0xb5, 0xe7, 0x42, 0x6e, 0x73, 0xfd, 0xe7, 0xd1, 0x41, 0xe1, 0xe3, 0xa7, 0x6, 0xcd, 0x45, 0x12, 0x17, 0xfd, 0xf9, 0x5d, 0xac, 0x5f, 0x59, 0xe9, 0xc7, 0x3c, 0xc9, 0xe, 0x33, 0x42, 0x70, 0x4f, 0x61, 0xd3, 0x13, 0xfb, 0x8f, 0x66, 0x11, 0xed, 0x89, 0x1f, 0x48, 0x72, 0x67, 0x60, 0x8, 0x6b, 0xc7, 0x33, 0x38, 0xe6, 0xd3, 0x4f, 0xfa, 0xae, 0x12, 0xa7, 0x2, 0x56, 0xcf, 0xf, 0xff, 0x9e, 0x87, 0x3b, 0xfa, 0xfc, 0x57, 0xac, 0x6, 0xa7, 0xa0, 0x8c, 0x99, 0xa4, 0x3e, 0xf4, 0x37, 0xcc, 0x7d, 0xb5, 0x53, 0x4c, 0x92, 0x7d, 0x4a, 0xb2, 0x32, 0x13, 0x73, 0xc9, 0x40, 0xdc, 0xe, 0x42, 0xa8, 0xd8, 0x69, 0xfb, 0x1a, 0x77, 0xb0, 0xe7, 0x97, 0x7a, 0x23, 0xda, 0x9b, 0x7e, 0x16, 0x60, 0x1d, 0x60, 0xa7, 0x29, 0xfe, 0x42, 0x10, 0xdc, 0xed, 0x52, 0x8d, 0xf5, 0x50, 0xdf, 0x2c, 0x58, 0xb7, 0x1f, 0xde, 0x8e, 0x56, 0x17, 0xae, 0x3f, 0x92, 0x5e, 0x9d, 0x13, 0xac, 0x7c, 0xbf, 0xcd, 0x5b, 0xe5, 0x2, 0x3, 0x1, 0x0, 0x1, 0xa3, 0x81, 0x8d, 0x30, 0x81, 0x8a, 0x30, 0xe, 0x6, 0x3, 0x55, 0x1d, 0xf, 0x1, 0x1, 0xff, 0x4, 0x4, 0x3, 0x2, 0x5, 0xa0, 0x30, 0x1d, 0x6, 0x3, 0x55, 0x1d, 0x25, 0x4, 0x16, 0x30, 0x14, 0x6, 0x8, 0x2b, 0x6, 0x1, 0x5, 0x5, 0x7, 0x3, 0x1, 0x6, 0x8, 0x2b, 0x6, 0x1, 0x5, 0x5, 0x7, 0x3, 0x2, 0x30, 0xc, 0x6, 0x3, 0x55, 0x1d, 0x13, 0x1, 0x1, 0xff, 0x4, 0x2, 0x30, 0x0, 0x30, 0x1f, 0x6, 0x3, 0x55, 0x1d, 0x23, 0x4, 0x18, 0x30, 0x16, 0x80, 0x14, 0x5c, 0xbf, 0x1d, 0x68, 0x28, 0xab, 0xb8, 0xaf, 0x7b, 0x88, 0x65, 0xa0, 0xf9, 0x81, 0xfa, 0x91, 0xa0, 0x7d, 0x8c, 0x74, 0x30, 0x2a, 0x6, 0x3, 0x55, 0x1d, 0x11, 0x4, 0x23, 0x30, 0x21, 0x82, 0x1f, 0x2a, 0x2e, 0x64, 0x65, 0x66, 0x61, 0x75, 0x6c, 0x74, 0x2e, 0x68, 0x75, 0x62, 0x62, 0x6c, 0x65, 0x2d, 0x67, 0x72, 0x70, 0x63, 0x2e, 0x63, 0x69, 0x6c, 0x69, 0x75, 0x6d, 0x2e, 0x69, 0x6f, 0x30, 0xd, 0x6, 0x9, 0x2a, 0x86, 0x48, 0x86, 0xf7, 0xd, 0x1, 0x1, 0xb, 0x5, 0x0, 0x3, 0x82, 0x1, 0x1, 0x0, 0x22, 0xf9, 0xa2, 0x3a, 0x5d, 0x3a, 0x38, 0xf3, 0xd7, 0x45, 0x37, 0xd1, 0xfb, 0x57, 0x48, 0xbc, 0x67, 0xa2, 0x1b, 0x70, 0xdb, 0x2, 0xad, 0x2, 0x70, 0xe1, 0xd4, 0x3a, 0x21, 0xf6, 0xf2, 0xa8, 0xec, 0xcd, 0xf0, 0xee, 0xb3, 0x58, 0x3b, 0xd1, 0x16, 0xac, 0xf4, 0xaa, 0x64, 0xdc, 0xd, 0xa, 0x37, 0x43, 0xf4, 0xb3, 0x8a, 0xed, 0xf, 0xc7, 0xa3, 0x95, 0x23, 0x6c, 0xce, 0xb3, 0x3d, 0xad, 0xc1, 0xa2, 0x8d, 0x5c, 0x89, 0xd0, 0x29, 0x59, 0x45, 0xf7, 0xb, 0xcd, 0xf3, 0xd4, 0xac, 0x97, 0x6d, 0x38, 0x58, 0xaf, 0x22, 0xbd, 0xbf, 0x56, 0x74, 0x4c, 0x18, 0xc9, 0x78, 0x60, 0xe1, 0xb8, 0x93, 0x6c, 0xf4, 0xf7, 0x1d, 0x68, 0x31, 0x47, 0x22, 0x42, 0xb, 0xa0, 0x4f, 0xfa, 0x14, 0xcb, 0x1c, 0x11, 0x34, 0x62, 0xe, 0xe3, 0xd9, 0x18, 0x9b, 0x9b, 0x19, 0x7, 0x30, 0x76, 0x81, 0xe, 0xab, 0x85, 0x30, 0x44, 0x9a, 0x1b, 0xd6, 0x99, 0xe4, 0xd1, 0x66, 0x21, 0xde, 0x7a, 0x39, 0xa2, 0x43, 0xcf, 0x3f, 0x30, 0x81, 0x3a, 0x79, 0xeb, 0x82, 0x72, 0x3d, 0xff, 0xce, 0xc3, 0x80, 0xbd, 0x98, 0x2a, 0xde, 0x78, 0x4a, 0x99, 0xa6, 0x16, 0x72, 0xb8, 0xb6, 0x9b, 0xcb, 0x43, 0xee, 0x98, 0x1e, 0x56, 0x3c, 0x59, 0x8c, 0xd1, 0xc4, 0x4, 0x48, 0xbf, 0x12, 0xad, 0x5e, 0xc5, 0xca, 0x28, 0xc7, 0x48, 0xf7, 0x2e, 0x8a, 0x86, 0xc2, 0xb1, 0x23, 0x34, 0x48, 0x64, 0x79, 0xbb, 0xec, 0xd1, 0xe1, 0x3d, 0x2b, 0x31, 0x33, 0xcd, 0xd8, 0x23, 0xda, 0x96, 0x8, 0x2c, 0x12, 0xeb, 0x83, 0x7a, 0x8, 0x2f, 0x4f, 0x30, 0x5, 0xc3, 0xc3, 0x6b, 0x5d, 0x7c, 0x65, 0x2b, 0x75, 0x52, 0xcc, 0xe, 0x77, 0xd3, 0x25, 0x39, 0xb0, 0x8e, 0x23, 0x6e, 0x9c, 0x4, 0x4c, 0x60, 0xa, 0xfe, 0x85, 0xa4}}, PrivateKey:(*rsa.PrivateKey)(0xc0000b11a0), SupportedSignatureAlgorithms:[]tls.SignatureScheme(nil), OCSPStaple:[]uint8(nil), SignedCertificateTimestamps:[][]uint8(nil), Leaf:(*x509.Certificate)(nil)}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -3,58 +3,58 @@
        	            	   Certificate: ([][]uint8) (len=1) {
        	            	-   ([]uint8) (len=869) {
        	            	-    00000000  30 82 03 61 30 82 02 49  a0 03 02 01 02 02 11 00  |0..a0..I........|
        	            	-    00000010  a5 30 7f 7f 9a 6b 02 11  e1 ab 8a d8 21 ea e5 bb  |.0...k......!...|
        	            	-    00000020  30 0d 06 09 2a 86 48 86  f7 0d 01 01 0b 05 00 30  |0...*.H........0|
        	            	-    00000030  1e 31 1c 30 1a 06 03 55  04 03 13 13 68 75 62 62  |.1.0...U....hubb|
        	            	-    00000040  6c 65 2d 63 61 2e 63 69  6c 69 75 6d 2e 69 6f 30  |le-ca.cilium.io0|
        	            	-    00000050  1e 17 0d 32 30 31 30 30  32 31 34 33 34 32 39 5a  |...201002143429Z|
        	            	-    00000060  17 0d 32 33 31 30 30 32  31 34 33 34 32 39 5a 30  |..231002143429Z0|
        	            	-    00000070  2a 31 28 30 26 06 03 55  04 03 0c 1f 2a 2e 64 65  |*1(0&..U....*.de|
        	            	-    00000080  66 61 75 6c 74 2e 68 75  62 62 6c 65 2d 67 72 70  |fault.hubble-grp|
        	            	-    00000090  63 2e 63 69 6c 69 75 6d  2e 69 6f 30 82 01 22 30  |c.cilium.io0.."0|
        	            	-    000000a0  0d 06 09 2a 86 48 86 f7  0d 01 01 01 05 00 03 82  |...*.H..........|
        	            	-    000000b0  01 0f 00 30 82 01 0a 02  82 01 01 00 cd 27 8e 63  |...0.........'.c|
        	            	-    000000c0  b1 25 2e 15 f3 b0 e0 a6  3f 46 9d 5a db 85 ff e8  |.%......?F.Z....|
        	            	-    000000d0  16 c3 4e 36 4e 62 1d e2  97 79 02 7d 45 e7 fa 46  |..N6Nb...y.}E..F|
        	            	-    000000e0  cf e5 1b c6 09 5f 61 08  93 6f 7b 7c e8 79 05 e2  |....._a..o{|.y..|
        	            	-    000000f0  ce e8 06 33 b5 2f b0 19  d0 a3 45 0e 6b c5 1a ee  |...3./....E.k...|
        	            	-    00000100  e2 91 fc 0f c5 57 09 63  17 b0 3c 80 a3 2b 90 55  |.....W.c..<..+.U|
        	            	-    00000110  11 1e 2f 93 35 48 ef c3  eb c2 97 e6 a6 67 22 a7  |../.5H.......g".|
        	            	-    00000120  92 ce b9 d8 43 ac 08 22  20 41 fc ab 5f bf 0f cd  |....C.." A.._...|
        	            	-    00000130  85 d2 fb 27 cd c2 13 8a  8f 3e 11 dd b1 fc 89 4e  |...'.....>.....N|
        	            	-    00000140  d3 5b 0b 63 dd 29 3f 32  d7 16 c3 5d ae f2 c5 29  |.[.c.)?2...]...)|
        	            	-    00000150  0b b1 eb c4 2b 3d f4 a7  a8 24 dd 0d 41 a4 28 0d  |....+=...$..A.(.|
        	            	-    00000160  f7 e1 8f 6d 70 3c 2b ca  52 f1 29 42 90 df 7b 2e  |...mp<+.R.)B..{.|
        	            	-    00000170  96 4d 2f 0b 5e bb 31 dd  14 80 93 cb 6b 76 e5 67  |.M/.^.1.....kv.g|
        	            	-    00000180  64 37 6f 3c 91 15 93 99  67 ec 13 b3 c2 22 2e a6  |d7o<....g...."..|
        	            	-    00000190  36 d1 e4 36 3c d7 63 1a  8f 73 33 9d d5 11 0b 58  |6..6<.c..s3....X|
        	            	-    000001a0  0b dd 62 2b 95 e7 ce f8  57 ff 36 f3 3e 9a c8 0e  |..b+....W.6.>...|
        	            	-    000001b0  10 af 6a fc 20 a6 13 ea  dc f2 1a d1 02 03 01 00  |..j. ...........|
        	            	-    000001c0  01 a3 81 8d 30 81 8a 30  0e 06 03 55 1d 0f 01 01  |....0..0...U....|
        	            	-    000001d0  ff 04 04 03 02 05 a0 30  1d 06 03 55 1d 25 04 16  |.......0...U.%..|
        	            	-    000001e0  30 14 06 08 2b 06 01 05  05 07 03 01 06 08 2b 06  |0...+.........+.|
        	            	-    000001f0  01 05 05 07 03 02 30 0c  06 03 55 1d 13 01 01 ff  |......0...U.....|
        	            	-    00000200  04 02 30 00 30 1f 06 03  55 1d 23 04 18 30 16 80  |..0.0...U.#..0..|
        	            	-    00000210  14 4e 89 ea ff 0c 9d 2a  66 70 40 38 23 8f 8a 86  |.N.....*fp@8#...|
        	            	-    00000220  36 6e b8 9a c1 30 2a 06  03 55 1d 11 04 23 30 21  |6n...0*..U...#0!|
        	            	-    00000230  82 1f 2a 2e 64 65 66 61  75 6c 74 2e 68 75 62 62  |..*.default.hubb|
        	            	-    00000240  6c 65 2d 67 72 70 63 2e  63 69 6c 69 75 6d 2e 69  |le-grpc.cilium.i|
        	            	-    00000250  6f 30 0d 06 09 2a 86 48  86 f7 0d 01 01 0b 05 00  |o0...*.H........|
        	            	-    00000260  03 82 01 01 00 a7 22 11  7c e0 5e 8b 54 9a 71 fd  |......".|.^.T.q.|
        	            	-    00000270  8c 4a 61 f0 b6 ed ec ae  0f ca 6b b7 a6 f0 53 ef  |.Ja.......k...S.|
        	            	-    00000280  65 ad 17 e2 94 2e f3 b9  37 c5 9d 2e cb 9d 24 79  |e.......7.....$y|
        	            	-    00000290  70 4e 19 ae e8 af 1e 01  d0 b5 70 9d 27 87 ff e4  |pN........p.'...|
        	            	-    000002a0  4c c7 0d ce 9a ba ea 2c  98 6b f2 3b 12 7c 4d 31  |L......,.k.;.|M1|
        	            	-    000002b0  b6 29 91 12 17 e5 d0 99  6e bc 58 29 68 2d c6 53  |.)......n.X)h-.S|
        	            	-    000002c0  e9 c3 2f 8c 13 d3 58 29  5e b2 b4 5f 83 c9 81 41  |../...X)^.._...A|
        	            	-    000002d0  58 c5 1a d7 f1 3e ea a2  d6 b4 2e 7a b6 94 3f b4  |X....>.....z..?.|
        	            	-    000002e0  f2 6e d4 2d 69 2d f4 52  61 21 1f 56 f9 8d 86 b1  |.n.-i-.Ra!.V....|
        	            	-    000002f0  96 b2 57 cf 97 d2 28 54  17 06 47 db 8e 02 08 8f  |..W...(T..G.....|
        	            	-    00000300  a3 72 b9 84 11 2e 83 88  fa b1 b1 0a d5 6f 54 d5  |.r...........oT.|
        	            	-    00000310  47 e2 81 48 01 41 14 7e  70 58 99 34 d5 bf ff 1d  |G..H.A.~pX.4....|
        	            	-    00000320  0e 93 4c 73 73 b1 14 43  dd f8 2c 65 87 ba f7 90  |..Lss..C..,e....|
        	            	-    00000330  01 7c 49 c9 d3 56 2f b3  75 5a 64 fc 79 78 00 45  |.|I..V/.uZd.yx.E|
        	            	-    00000340  bb 15 c6 46 14 11 d5 0a  fe 82 17 36 6e 2e b2 36  |...F.......6n..6|
        	            	-    00000350  e0 f8 1a b1 5a f6 30 52  02 9d 54 78 a9 9f 57 c0  |....Z.0R..Tx..W.|
        	            	-    00000360  93 57 bf 9d 96                                    |.W...|
        	            	+   ([]uint8) (len=868) {
        	            	+    00000000  30 82 03 60 30 82 02 48  a0 03 02 01 02 02 10 71  |0..`0..H.......q|
        	            	+    00000010  00 b4 e2 8c 4f 26 5e dc  a2 d5 5a 49 ec e5 cf 30  |....O&^...ZI...0|
        	            	+    00000020  0d 06 09 2a 86 48 86 f7  0d 01 01 0b 05 00 30 1e  |...*.H........0.|
        	            	+    00000030  31 1c 30 1a 06 03 55 04  03 13 13 68 75 62 62 6c  |1.0...U....hubbl|
        	            	+    00000040  65 2d 63 61 2e 63 69 6c  69 75 6d 2e 69 6f 30 1e  |e-ca.cilium.io0.|
        	            	+    00000050  17 0d 32 30 31 30 30 31  31 33 32 35 33 31 5a 17  |..201001132531Z.|
        	            	+    00000060  0d 32 33 31 30 30 31 31  33 32 35 33 31 5a 30 2a  |.231001132531Z0*|
        	            	+    00000070  31 28 30 26 06 03 55 04  03 0c 1f 2a 2e 64 65 66  |1(0&..U....*.def|
        	            	+    00000080  61 75 6c 74 2e 68 75 62  62 6c 65 2d 67 72 70 63  |ault.hubble-grpc|
        	            	+    00000090  2e 63 69 6c 69 75 6d 2e  69 6f 30 82 01 22 30 0d  |.cilium.io0.."0.|
        	            	+    000000a0  06 09 2a 86 48 86 f7 0d  01 01 01 05 00 03 82 01  |..*.H...........|
        	            	+    000000b0  0f 00 30 82 01 0a 02 82  01 01 00 da 0f c9 0d 9f  |..0.............|
        	            	+    000000c0  31 b9 5c 71 64 06 79 7f  f3 9b 77 2c 7d 6e 42 27  |1.\qd.y...w,}nB'|
        	            	+    000000d0  dd 32 62 d2 73 95 f2 6a  24 62 b3 30 4c 6a c1 0c  |.2b.s..j$b.0Lj..|
        	            	+    000000e0  ee 51 25 21 aa f1 9d 3f  06 5d b7 1a c0 00 39 11  |.Q%!...?.]....9.|
        	            	+    000000f0  3a 20 d8 c6 9c b4 7d 17  16 1a b0 3e 6d f1 ae f7  |: ....}....>m...|
        	            	+    00000100  09 ab 94 18 8a 84 9d 1e  6f 27 2f 3b ac 34 ce eb  |........o'/;.4..|
        	            	+    00000110  2e fe bc 30 a3 f7 1a e0  44 38 cd 61 70 07 b3 89  |...0....D8.ap...|
        	            	+    00000120  4d 71 1a 87 30 23 c2 6d  f1 b5 e7 42 6e 73 fd e7  |Mq..0#.m...Bns..|
        	            	+    00000130  d1 41 e1 e3 a7 06 cd 45  12 17 fd f9 5d ac 5f 59  |.A.....E....]._Y|
        	            	+    00000140  e9 c7 3c c9 0e 33 42 70  4f 61 d3 13 fb 8f 66 11  |..<..3BpOa....f.|
        	            	+    00000150  ed 89 1f 48 72 67 60 08  6b c7 33 38 e6 d3 4f fa  |...Hrg`.k.38..O.|
        	            	+    00000160  ae 12 a7 02 56 cf 0f ff  9e 87 3b fa fc 57 ac 06  |....V.....;..W..|
        	            	+    00000170  a7 a0 8c 99 a4 3e f4 37  cc 7d b5 53 4c 92 7d 4a  |.....>.7.}.SL.}J|
        	            	+    00000180  b2 32 13 73 c9 40 dc 0e  42 a8 d8 69 fb 1a 77 b0  |.2.s.@..B..i..w.|
        	            	+    00000190  e7 97 7a 23 da 9b 7e 16  60 1d 60 a7 29 fe 42 10  |..z#..~.`.`.).B.|
        	            	+    000001a0  dc ed 52 8d f5 50 df 2c  58 b7 1f de 8e 56 17 ae  |..R..P.,X....V..|
        	            	+    000001b0  3f 92 5e 9d 13 ac 7c bf  cd 5b e5 02 03 01 00 01  |?.^...|..[......|
        	            	+    000001c0  a3 81 8d 30 81 8a 30 0e  06 03 55 1d 0f 01 01 ff  |...0..0...U.....|
        	            	+    000001d0  04 04 03 02 05 a0 30 1d  06 03 55 1d 25 04 16 30  |......0...U.%..0|
        	            	+    000001e0  14 06 08 2b 06 01 05 05  07 03 01 06 08 2b 06 01  |...+.........+..|
        	            	+    000001f0  05 05 07 03 02 30 0c 06  03 55 1d 13 01 01 ff 04  |.....0...U......|
        	            	+    00000200  02 30 00 30 1f 06 03 55  1d 23 04 18 30 16 80 14  |.0.0...U.#..0...|
        	            	+    00000210  5c bf 1d 68 28 ab b8 af  7b 88 65 a0 f9 81 fa 91  |\..h(...{.e.....|
        	            	+    00000220  a0 7d 8c 74 30 2a 06 03  55 1d 11 04 23 30 21 82  |.}.t0*..U...#0!.|
        	            	+    00000230  1f 2a 2e 64 65 66 61 75  6c 74 2e 68 75 62 62 6c  |.*.default.hubbl|
        	            	+    00000240  65 2d 67 72 70 63 2e 63  69 6c 69 75 6d 2e 69 6f  |e-grpc.cilium.io|
        	            	+    00000250  30 0d 06 09 2a 86 48 86  f7 0d 01 01 0b 05 00 03  |0...*.H.........|
        	            	+    00000260  82 01 01 00 22 f9 a2 3a  5d 3a 38 f3 d7 45 37 d1  |...."..:]:8..E7.|
        	            	+    00000270  fb 57 48 bc 67 a2 1b 70  db 02 ad 02 70 e1 d4 3a  |.WH.g..p....p..:|
        	            	+    00000280  21 f6 f2 a8 ec cd f0 ee  b3 58 3b d1 16 ac f4 aa  |!........X;.....|
        	            	+    00000290  64 dc 0d 0a 37 43 f4 b3  8a ed 0f c7 a3 95 23 6c  |d...7C........#l|
        	            	+    000002a0  ce b3 3d ad c1 a2 8d 5c  89 d0 29 59 45 f7 0b cd  |..=....\..)YE...|
        	            	+    000002b0  f3 d4 ac 97 6d 38 58 af  22 bd bf 56 74 4c 18 c9  |....m8X."..VtL..|
        	            	+    000002c0  78 60 e1 b8 93 6c f4 f7  1d 68 31 47 22 42 0b a0  |x`...l...h1G"B..|
        	            	+    000002d0  4f fa 14 cb 1c 11 34 62  0e e3 d9 18 9b 9b 19 07  |O.....4b........|
        	            	+    000002e0  30 76 81 0e ab 85 30 44  9a 1b d6 99 e4 d1 66 21  |0v....0D......f!|
        	            	+    000002f0  de 7a 39 a2 43 cf 3f 30  81 3a 79 eb 82 72 3d ff  |.z9.C.?0.:y..r=.|
        	            	+    00000300  ce c3 80 bd 98 2a de 78  4a 99 a6 16 72 b8 b6 9b  |.....*.xJ...r...|
        	            	+    00000310  cb 43 ee 98 1e 56 3c 59  8c d1 c4 04 48 bf 12 ad  |.C...V<Y....H...|
        	            	+    00000320  5e c5 ca 28 c7 48 f7 2e  8a 86 c2 b1 23 34 48 64  |^..(.H......#4Hd|
        	            	+    00000330  79 bb ec d1 e1 3d 2b 31  33 cd d8 23 da 96 08 2c  |y....=+13..#...,|
        	            	+    00000340  12 eb 83 7a 08 2f 4f 30  05 c3 c3 6b 5d 7c 65 2b  |...z./O0...k]|e+|
        	            	+    00000350  75 52 cc 0e 77 d3 25 39  b0 8e 23 6e 9c 04 4c 60  |uR..w.%9..#n..L`|
        	            	+    00000360  0a fe 85 a4                                       |....|
        	            	    }
        	            	@@ -66,34 +66,34 @@
        	            	      abs: (big.nat) (len=32) {
        	            	-      (big.Word) 2352589754797791953,
        	            	-      (big.Word) 4511137939500002044,
        	            	-      (big.Word) 10801829797294978803,
        	            	-      (big.Word) 15353065077466817067,
        	            	-      (big.Word) 4384081727995458461,
        	            	-      (big.Word) 13988794683985355830,
        	            	-      (big.Word) 10454424397093278643,
        	            	-      (big.Word) 7743628841535893308,
        	            	-      (big.Word) 6826104485811622859,
        	            	-      (big.Word) 10439197901288451851,
        	            	-      (big.Word) 8087387178825492802,
        	            	-      (big.Word) 4729949549104303981,
        	            	-      (big.Word) 3115915518127955213,
        	            	-      (big.Word) 12606355087043980228,
        	            	-      (big.Word) 15936338244016718685,
        	            	-      (big.Word) 12825276810490743651,
        	            	-      (big.Word) 14826434409086718429,
        	            	-      (big.Word) 6899250529542798119,
        	            	-      (big.Word) 4876281439198510251,
        	            	-      (big.Word) 11990590636015139288,
        	            	-      (big.Word) 3839582307086407654,
        	            	-      (big.Word) 11757649947214622611,
        	            	-      (big.Word) 14219844669649468544,
        	            	-      (big.Word) 7765642745776045071,
        	            	-      (big.Word) 13055847469691389198,
        	            	-      (big.Word) 16751426760673330739,
        	            	-      (big.Word) 675365158590053244,
        	            	-      (big.Word) 5037269890279742406,
        	            	-      (big.Word) 5648109741717127805,
        	            	-      (big.Word) 15818330638488129078,
        	            	-      (big.Word) 17559781951748349274,
        	            	-      (big.Word) 14782940860691852821
        	            	+      (big.Word) 11318579940306017253,
        	            	+      (big.Word) 16036849982820553310,
        	            	+      (big.Word) 10229171048093890335,
        	            	+      (big.Word) 12045438238083312978,
        	            	+      (big.Word) 2583548302093983072,
        	            	+      (big.Word) 7636726694508205946,
        	            	+      (big.Word) 8343271098556786904,
        	            	+      (big.Word) 6002333470199788051,
        	            	+      (big.Word) 11071043002612612533,
        	            	+      (big.Word) 18085426599970709644,
        	            	+      (big.Word) 168549703693010747,
        	            	+      (big.Word) 4100196851229004455,
        	            	+      (big.Word) 5220348580231628595,
        	            	+      (big.Word) 1439902174425221407,
        	            	+      (big.Word) 14487573411742573011,
        	            	+      (big.Word) 17968707613808510780,
        	            	+      (big.Word) 16404087646417197053,
        	            	+      (big.Word) 4786890988294586849,
        	            	+      (big.Word) 9741325311977567719,
        	            	+      (big.Word) 7021120286754697498,
        	            	+      (big.Word) 3504916629828155597,
        	            	+      (big.Word) 4299869807545614012,
        	            	+      (big.Word) 1768371614051673903,
        	            	+      (big.Word) 4498517336584203156,
        	            	+      (big.Word) 14311512165227829936,
        	            	+      (big.Word) 1927540885616730328,
        	            	+      (big.Word) 2426017006961974711,
        	            	+      (big.Word) 3480273989428728101,
        	            	+      (big.Word) 15164629238681985715,
        	            	+      (big.Word) 3205839740173365858,
        	            	+      (big.Word) 8170662742864403319,
        	            	+      (big.Word) 15712998685261543772
        	            	      }
        	            	@@ -105,34 +105,34 @@
        	            	     abs: (big.nat) (len=32) {
        	            	-     (big.Word) 15624741362008875937,
        	            	-     (big.Word) 994519418029673208,
        	            	-     (big.Word) 4963449718631793190,
        	            	-     (big.Word) 7653280040807305766,
        	            	-     (big.Word) 15679776892366813907,
        	            	-     (big.Word) 12400418795957161298,
        	            	-     (big.Word) 5165742679402821579,
        	            	-     (big.Word) 9289675929270772130,
        	            	-     (big.Word) 7899131558397467609,
        	            	-     (big.Word) 3388311456638923001,
        	            	-     (big.Word) 6368371093757320879,
        	            	-     (big.Word) 5693291147932139800,
        	            	-     (big.Word) 18343679879163238065,
        	            	-     (big.Word) 5793704278097547013,
        	            	-     (big.Word) 13676256617794488566,
        	            	-     (big.Word) 17891876012011448093,
        	            	-     (big.Word) 9481472548479742621,
        	            	-     (big.Word) 16512981926342100387,
        	            	-     (big.Word) 17928947998674368859,
        	            	-     (big.Word) 8213443683689342812,
        	            	-     (big.Word) 9271927355867284876,
        	            	-     (big.Word) 17434318681068750257,
        	            	-     (big.Word) 266168980739669794,
        	            	-     (big.Word) 6006204822093897754,
        	            	-     (big.Word) 9546354247988640954,
        	            	-     (big.Word) 6247510887747997988,
        	            	-     (big.Word) 4123948179430180305,
        	            	-     (big.Word) 2025254991956869301,
        	            	-     (big.Word) 12020854673600872424,
        	            	-     (big.Word) 4022241529349743654,
        	            	-     (big.Word) 11130273803261016473,
        	            	-     (big.Word) 1185576653856544828
        	            	+     (big.Word) 8860123264223949857,
        	            	+     (big.Word) 17304568639182079333,
        	            	+     (big.Word) 7418949273586417981,
        	            	+     (big.Word) 4584086378004883160,
        	            	+     (big.Word) 6263405951468740764,
        	            	+     (big.Word) 4106372815932006979,
        	            	+     (big.Word) 12762456719949744522,
        	            	+     (big.Word) 5658381549108389604,
        	            	+     (big.Word) 4322766621912408047,
        	            	+     (big.Word) 13565287000842320975,
        	            	+     (big.Word) 17681427532035818516,
        	            	+     (big.Word) 15965837137149612692,
        	            	+     (big.Word) 13252877325718328082,
        	            	+     (big.Word) 11128060174877128903,
        	            	+     (big.Word) 3190785248185855207,
        	            	+     (big.Word) 8553813904659145358,
        	            	+     (big.Word) 3491595176975165282,
        	            	+     (big.Word) 14928578528403344865,
        	            	+     (big.Word) 6338924352487631916,
        	            	+     (big.Word) 3441435248265286640,
        	            	+     (big.Word) 9696902943269357597,
        	            	+     (big.Word) 4200966186567641668,
        	            	+     (big.Word) 9195470738873123648,
        	            	+     (big.Word) 14815721956816291931,
        	            	+     (big.Word) 15307178224072337216,
        	            	+     (big.Word) 2096082236328571412,
        	            	+     (big.Word) 10647364162644695730,
        	            	+     (big.Word) 17426154521693406685,
        	            	+     (big.Word) 7551708630688556980,
        	            	+     (big.Word) 6398734182385868644,
        	            	+     (big.Word) 6354962659511472362,
        	            	+     (big.Word) 1931727579949528635
        	            	     }
        	            	@@ -143,18 +143,18 @@
        	            	      abs: (big.nat) (len=16) {
        	            	-      (big.Word) 5433252485841197571,
        	            	-      (big.Word) 11000973859376617606,
        	            	-      (big.Word) 9482807978636476513,
        	            	-      (big.Word) 8674424281312371255,
        	            	-      (big.Word) 12572944811230479056,
        	            	-      (big.Word) 10832129783537163017,
        	            	-      (big.Word) 1569554778853405276,
        	            	-      (big.Word) 485271073230948116,
        	            	-      (big.Word) 11229791628926896821,
        	            	-      (big.Word) 4246995376321346230,
        	            	-      (big.Word) 2721931763854940053,
        	            	-      (big.Word) 5196274700396993551,
        	            	-      (big.Word) 6095555133869312207,
        	            	-      (big.Word) 138749479983938176,
        	            	-      (big.Word) 2708017787025654049,
        	            	-      (big.Word) 16246650950038953514
        	            	+      (big.Word) 5928010806743201369,
        	            	+      (big.Word) 10601600486882917768,
        	            	+      (big.Word) 5157642079648974585,
        	            	+      (big.Word) 7909800316825155056,
        	            	+      (big.Word) 11685709632771566394,
        	            	+      (big.Word) 1482266349414007127,
        	            	+      (big.Word) 274510651177520381,
        	            	+      (big.Word) 11084178815497944796,
        	            	+      (big.Word) 15601622175845714769,
        	            	+      (big.Word) 3997380225501664627,
        	            	+      (big.Word) 981524718530509504,
        	            	+      (big.Word) 6459553128542002716,
        	            	+      (big.Word) 13509220441648012286,
        	            	+      (big.Word) 766035841968012621,
        	            	+      (big.Word) 10409789110322607543,
        	            	+      (big.Word) 17762382429307082322
        	            	      }
        	            	@@ -164,18 +164,18 @@
        	            	      abs: (big.nat) (len=16) {
        	            	-      (big.Word) 3633027726856440219,
        	            	-      (big.Word) 7677598162804246813,
        	            	-      (big.Word) 11342648504415138955,
        	            	-      (big.Word) 10604794515381100814,
        	            	-      (big.Word) 10297863506438175216,
        	            	-      (big.Word) 5617600569469106365,
        	            	-      (big.Word) 14406662699045392323,
        	            	-      (big.Word) 3141314243500939177,
        	            	-      (big.Word) 2556726525084129136,
        	            	-      (big.Word) 1542316932863857850,
        	            	-      (big.Word) 7943843008250915628,
        	            	-      (big.Word) 8673018833645310211,
        	            	-      (big.Word) 8167595727695779842,
        	            	-      (big.Word) 13233810505771448836,
        	            	-      (big.Word) 15946318506863058405,
        	            	-      (big.Word) 16784820918018946331
        	            	+      (big.Word) 18161656390350223469,
        	            	+      (big.Word) 10433028609059336137,
        	            	+      (big.Word) 3734890419699341475,
        	            	+      (big.Word) 6378931234074726188,
        	            	+      (big.Word) 13082406543347939112,
        	            	+      (big.Word) 1440341939464900114,
        	            	+      (big.Word) 14195436738311231615,
        	            	+      (big.Word) 8898217189632600209,
        	            	+      (big.Word) 14871387894456180116,
        	            	+      (big.Word) 10968982738594964282,
        	            	+      (big.Word) 17523853375580147287,
        	            	+      (big.Word) 6540245292770377236,
        	            	+      (big.Word) 1001132373665138113,
        	            	+      (big.Word) 1957342899235883805,
        	            	+      (big.Word) 5453518594272705309,
        	            	+      (big.Word) 16318400221994413315
        	            	      }
        	            	@@ -187,18 +187,18 @@
        	            	      abs: (big.nat) (len=16) {
        	            	-      (big.Word) 17799342123166640051,
        	            	-      (big.Word) 7381972448863883428,
        	            	-      (big.Word) 1491177704679870199,
        	            	-      (big.Word) 6726659754418321244,
        	            	-      (big.Word) 35950775990839330,
        	            	-      (big.Word) 17312618922151741822,
        	            	-      (big.Word) 2790359673119955866,
        	            	-      (big.Word) 13852862755539058018,
        	            	-      (big.Word) 8552499384911887599,
        	            	-      (big.Word) 4947208590561876107,
        	            	-      (big.Word) 923782072731271048,
        	            	-      (big.Word) 9222145988473004793,
        	            	-      (big.Word) 17109726296784155064,
        	            	-      (big.Word) 5618734154945337409,
        	            	-      (big.Word) 18054486555472786962,
        	            	-      (big.Word) 9065471086439331843
        	            	+      (big.Word) 15042280822238168753,
        	            	+      (big.Word) 9950222982275874671,
        	            	+      (big.Word) 16853625035045647765,
        	            	+      (big.Word) 7767864527403976710,
        	            	+      (big.Word) 12487357700293653596,
        	            	+      (big.Word) 9237763228492907203,
        	            	+      (big.Word) 13122120580181292623,
        	            	+      (big.Word) 6950991483022142060,
        	            	+      (big.Word) 4407975771614070036,
        	            	+      (big.Word) 9742119585733461492,
        	            	+      (big.Word) 12655065647746570415,
        	            	+      (big.Word) 18179598786713871378,
        	            	+      (big.Word) 16861197967363437267,
        	            	+      (big.Word) 17193747228364294277,
        	            	+      (big.Word) 2965778729830205983,
        	            	+      (big.Word) 8447952459244423089
        	            	      }
        	            	@@ -208,18 +208,18 @@
        	            	      abs: (big.nat) (len=16) {
        	            	-      (big.Word) 4655001799382403643,
        	            	-      (big.Word) 12370217941278030090,
        	            	-      (big.Word) 6464540753339837098,
        	            	-      (big.Word) 1066760982279285045,
        	            	-      (big.Word) 7671024008138615353,
        	            	-      (big.Word) 4114612796870183951,
        	            	-      (big.Word) 5205998992562009359,
        	            	-      (big.Word) 2915292608539390892,
        	            	-      (big.Word) 3222507947170979480,
        	            	-      (big.Word) 15923164127135176743,
        	            	-      (big.Word) 3127705078159264537,
        	            	-      (big.Word) 16576179577875350607,
        	            	-      (big.Word) 17671284651649245942,
        	            	-      (big.Word) 5172771724865132089,
        	            	-      (big.Word) 2733195462199603690,
        	            	-      (big.Word) 6396912767892323764
        	            	+      (big.Word) 6996425599371487713,
        	            	+      (big.Word) 426692014020340188,
        	            	+      (big.Word) 4293825361674773428,
        	            	+      (big.Word) 9019553182030859695,
        	            	+      (big.Word) 4149683775255310711,
        	            	+      (big.Word) 8575017655253727135,
        	            	+      (big.Word) 10054566203367077537,
        	            	+      (big.Word) 3082620754560983855,
        	            	+      (big.Word) 7903506879137783469,
        	            	+      (big.Word) 11822980477172637810,
        	            	+      (big.Word) 17724362784225061087,
        	            	+      (big.Word) 4101827159155460789,
        	            	+      (big.Word) 9644016460977374830,
        	            	+      (big.Word) 12426254661124569809,
        	            	+      (big.Word) 16432169503445193872,
        	            	+      (big.Word) 440223562147887799
        	            	      }
        	            	@@ -229,18 +229,18 @@
        	            	      abs: (big.nat) (len=16) {
        	            	-      (big.Word) 11062494198099440237,
        	            	-      (big.Word) 7236832674609069939,
        	            	-      (big.Word) 17830637127224535467,
        	            	-      (big.Word) 2572802936289812105,
        	            	-      (big.Word) 16960114502625980056,
        	            	-      (big.Word) 5881217302955321232,
        	            	-      (big.Word) 3471011141926750247,
        	            	-      (big.Word) 6306539380935200532,
        	            	-      (big.Word) 5630044660026799123,
        	            	-      (big.Word) 9439257219852849140,
        	            	-      (big.Word) 7576717133247581560,
        	            	-      (big.Word) 11452727171223977150,
        	            	-      (big.Word) 5534454935637724478,
        	            	-      (big.Word) 3598525439675532243,
        	            	-      (big.Word) 9773862385303730161,
        	            	-      (big.Word) 7620631324252459287
        	            	+      (big.Word) 8964848867002177637,
        	            	+      (big.Word) 6460018337016804848,
        	            	+      (big.Word) 1685435000722037327,
        	            	+      (big.Word) 1214934668740900097,
        	            	+      (big.Word) 348105529637652699,
        	            	+      (big.Word) 6274238000556711738,
        	            	+      (big.Word) 5745603040390365407,
        	            	+      (big.Word) 6324538532017452836,
        	            	+      (big.Word) 8875408564989861372,
        	            	+      (big.Word) 8181700612891890568,
        	            	+      (big.Word) 12237274858146655743,
        	            	+      (big.Word) 3493602050220924776,
        	            	+      (big.Word) 6090982364501268931,
        	            	+      (big.Word) 3482876329915161714,
        	            	+      (big.Word) 6314803407091111621,
        	            	+      (big.Word) 16604883304989885147
        	            	      }
        	Test:       	TestWatchedServerConfigRotation
```
</details>

[96865ea7_RuntimeDatapathPrivilegedUnitTests_Run_Tests.zip](https://github.com/cilium/cilium/files/9936565/96865ea7_RuntimeDatapathPrivilegedUnitTests_Run_Tests.zip)

Jenkins: https://jenkins.cilium.io/job/Cilium-PR-Runtime-net-next/3854/testReport/(root)/Suite-runtime/RuntimeDatapathPrivilegedUnitTests_Run_Tests/

--- a/pkg/crypto/certloader/client_test.go
+++ b/pkg/crypto/certloader/client_test.go
@@ -124,8 +124,20 @@ func TestWatchedClientConfigRotation(t *testing.T) {
 	assert.NotNil(t, c)
 	defer c.Stop()
 
+	prevKeypairGeneration, prevCaCertPoolGeneration := c.generations()
 	rotate(t, hubble, relay)
-	<-time.After(testReloadDelay)
+
+	// wait until both keypair and caCertPool have been reloaded
+	ticker := time.NewTicker(testReloadDelay)
+	defer ticker.Stop()
+	for range ticker.C {
+		keypairGeneration, caCertPoolGeneration := c.generations()
+		keypairUpdated := keypairGeneration > prevKeypairGeneration
+		caCertPoolUpdated := caCertPoolGeneration > prevCaCertPoolGeneration
+		if keypairUpdated && caCertPoolUpdated {
+			break
+		}
+	}
 
 	tlsConfig := c.ClientConfig(&tls.Config{
 		MinVersion: tls.VersionTLS13,

--- a/pkg/crypto/certloader/reloader.go
+++ b/pkg/crypto/certloader/reloader.go
@@ -25,8 +25,10 @@ type FileReloader struct {
 	mutex       lock.Mutex
 	// fields below should only be accessed with mutex acquired as they may be
 	// updated concurrently.
-	caCertPool *x509.CertPool
-	keypair    *tls.Certificate
+	caCertPool           *x509.CertPool
+	caCertPoolGeneration uint // incremented when caCertPool is reloaded
+	keypair              *tls.Certificate
+	keypairGeneration    uint // incremented when keypair is reloaded
 }
 
 var (
@@ -124,8 +126,14 @@ func (r *FileReloader) Reload() (keypair *tls.Certificate, caCertPool *x509.Cert
 	}
 
 	r.mutex.Lock()
-	r.keypair = keypair
-	r.caCertPool = caCertPool
+	if r.HasKeypair() {
+		r.keypair = keypair
+		r.keypairGeneration++
+	}
+	if r.HasCustomCA() {
+		r.caCertPool = caCertPool
+		r.caCertPoolGeneration++
+	}
 	r.mutex.Unlock()
 	return
 }
@@ -142,6 +150,7 @@ func (r *FileReloader) ReloadKeypair() (*tls.Certificate, error) {
 	}
 	r.mutex.Lock()
 	r.keypair = keypair
+	r.keypairGeneration++
 	r.mutex.Unlock()
 	return keypair, nil
 }
@@ -158,6 +167,7 @@ func (r *FileReloader) ReloadCA() (*x509.CertPool, error) {
 	}
 	r.mutex.Lock()
 	r.caCertPool = caCertPool
+	r.caCertPoolGeneration++
 	r.mutex.Unlock()
 	return caCertPool, nil
 }
@@ -184,4 +194,11 @@ func (r *FileReloader) readCertificateAuthority() (*x509.CertPool, error) {
 		}
 	}
 	return caCertPool, nil
+}
+
+// generations returns the keypair and caCertPool generation counters.
+func (r *FileReloader) generations() (uint, uint) {
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
+	return r.keypairGeneration, r.caCertPoolGeneration
 }

--- a/pkg/crypto/certloader/reloader_test.go
+++ b/pkg/crypto/certloader/reloader_test.go
@@ -449,15 +449,30 @@ func TestReload(t *testing.T) {
 				t.Error(err)
 				return
 			}
+			prevKeypairGeneration, prevCaCertPoolGeneration := r.generations()
 			keypair, caCertPool, err := r.Reload()
 			assert.Nil(t, err)
+			// keypair check
 			assert.Equal(t, tt.expectedKeypair, keypair)
+			// caCertPool check
 			if tt.expectedCaCertPool != nil {
 				assert.Equal(t, tt.expectedCaCertPool.Subjects(), caCertPool.Subjects())
 			} else {
 				assert.Nil(t, caCertPool)
 			}
-
+			// generations check
+			keypairGeneration, caCertPoolGeneration := r.generations()
+			if tt.expectedKeypair != nil {
+				assert.Equal(t, prevKeypairGeneration+1, keypairGeneration)
+			} else {
+				assert.Equal(t, prevKeypairGeneration, keypairGeneration)
+			}
+			if tt.expectedCaCertPool != nil {
+				assert.Equal(t, prevCaCertPoolGeneration+1, caCertPoolGeneration)
+			} else {
+				assert.Equal(t, prevCaCertPoolGeneration, caCertPoolGeneration)
+			}
+			// ensures that KeypairAndCACertPool() returns the expected values
 			keypair, caCertPool = r.KeypairAndCACertPool()
 			assert.Equal(t, tt.expectedKeypair, keypair)
 			if tt.expectedCaCertPool != nil {
@@ -480,10 +495,9 @@ func TestReloadKeypair(t *testing.T) {
 	}
 
 	tests := []struct {
-		name               string
-		constructor        func() (*FileReloader, error)
-		expectedKeypair    *tls.Certificate
-		expectedCaCertPool *x509.CertPool
+		name            string
+		constructor     func() (*FileReloader, error)
+		expectedKeypair *tls.Certificate
 	}{
 		{
 			name: "empty",
@@ -522,10 +536,19 @@ func TestReloadKeypair(t *testing.T) {
 				t.Error(err)
 				return
 			}
+			prevKeypairGeneration, _ := r.generations()
 			keypair, err := r.ReloadKeypair()
 			assert.Nil(t, err)
+			// keypair check
 			assert.Equal(t, tt.expectedKeypair, keypair)
-
+			// generations check
+			keypairGeneration, _ := r.generations()
+			if tt.expectedKeypair != nil {
+				assert.Equal(t, prevKeypairGeneration+1, keypairGeneration)
+			} else {
+				assert.Equal(t, prevKeypairGeneration, keypairGeneration)
+			}
+			// ensures that KeypairAndCACertPool() returns the expected values
 			keypair, caCertPool := r.KeypairAndCACertPool()
 			assert.Equal(t, tt.expectedKeypair, keypair)
 			assert.Nil(t, caCertPool)
@@ -585,14 +608,23 @@ func TestReloadCA(t *testing.T) {
 				t.Error(err)
 				return
 			}
+			_, prevCaCertPoolGeneration := r.generations()
 			caCertPool, err := r.ReloadCA()
 			assert.Nil(t, err)
+			// caCertPool check
 			if tt.expectedCaCertPool != nil {
 				assert.Equal(t, tt.expectedCaCertPool.Subjects(), caCertPool.Subjects())
 			} else {
 				assert.Nil(t, caCertPool)
 			}
-
+			// generations check
+			_, caCertPoolGeneration := r.generations()
+			if tt.expectedCaCertPool != nil {
+				assert.Equal(t, prevCaCertPoolGeneration+1, caCertPoolGeneration)
+			} else {
+				assert.Equal(t, prevCaCertPoolGeneration, caCertPoolGeneration)
+			}
+			// ensures that KeypairAndCACertPool() returns the expected values
 			keypair, caCertPool := r.KeypairAndCACertPool()
 			assert.Nil(t, keypair)
 			if tt.expectedCaCertPool != nil {
@@ -633,6 +665,7 @@ func TestReloadError(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	prevKeypairGeneration, prevCaCertPoolGeneration := r.generations()
 	_, _, err = r.Reload()
 	assert.NotNil(t, err)
 
@@ -640,4 +673,8 @@ func TestReloadError(t *testing.T) {
 	keypair, caCertPool = r.KeypairAndCACertPool()
 	assert.Equal(t, &expectedKeypair, keypair)
 	assert.Equal(t, expectedCaCertPool.Subjects(), caCertPool.Subjects())
+	// generations should not have changed
+	keypairGeneration, caCertPoolGeneration := r.generations()
+	assert.Equal(t, prevKeypairGeneration, keypairGeneration)
+	assert.Equal(t, prevCaCertPoolGeneration, caCertPoolGeneration)
 }

--- a/pkg/crypto/certloader/server_test.go
+++ b/pkg/crypto/certloader/server_test.go
@@ -148,8 +148,20 @@ func TestWatchedServerConfigRotation(t *testing.T) {
 	assert.NotNil(t, s)
 	defer s.Stop()
 
+	prevKeypairGeneration, prevCaCertPoolGeneration := s.generations()
 	rotate(t, hubble, relay)
-	<-time.After(testReloadDelay)
+
+	// wait until both keypair and caCertPool have been reloaded
+	ticker := time.NewTicker(testReloadDelay)
+	defer ticker.Stop()
+	for range ticker.C {
+		keypairGeneration, caCertPoolGeneration := s.generations()
+		keypairUpdated := keypairGeneration > prevKeypairGeneration
+		caCertPoolUpdated := caCertPoolGeneration > prevCaCertPoolGeneration
+		if keypairUpdated && caCertPoolUpdated {
+			break
+		}
+	}
 
 	generator := s.ServerConfig(&tls.Config{
 		MinVersion: tls.VersionTLS13,

--- a/pkg/crypto/certloader/server_test.go
+++ b/pkg/crypto/certloader/server_test.go
@@ -81,7 +81,6 @@ func TestFutureWatchedServerConfig(t *testing.T) {
 	assert.Nil(t, err)
 
 	// the files don't exists, expect the config to not be ready yet.
-	var s *WatchedServerConfig
 	select {
 	case <-ch:
 		t.Fatal("FutureWatchedServerConfig should not be ready without the TLS files")
@@ -91,11 +90,7 @@ func TestFutureWatchedServerConfig(t *testing.T) {
 	setup(t, hubble, relay)
 
 	// the files exists now, expect the watcher to become ready.
-	select {
-	case s = <-ch:
-	case <-time.After(testReloadDelay):
-		t.Fatal("FutureWatchedServerConfig should be ready one the TLS files exists")
-	}
+	s := <-ch
 	s.Stop()
 }
 

--- a/pkg/crypto/certloader/watcher_test.go
+++ b/pkg/crypto/certloader/watcher_test.go
@@ -92,12 +92,7 @@ func TestFutureWatcherImmediately(t *testing.T) {
 	assert.Nil(t, err)
 
 	// the files already exists, expect the watcher to be readily available.
-	var w *Watcher
-	select {
-	case w = <-ch:
-	case <-time.After(testReloadDelay):
-		t.Fatal("FutureWatcher should be ready immediately")
-	}
+	w := <-ch
 	defer w.Stop()
 
 	keypair, caCertPool := w.KeypairAndCACertPool()
@@ -135,11 +130,7 @@ func TestFutureWatcher(t *testing.T) {
 	setup(t, hubble, relay)
 
 	// the files exists now, expect the watcher to become ready.
-	select {
-	case w = <-ch:
-	case <-time.After(testReloadDelay):
-		t.Fatal("FutureWatcher should be ready once the TLS files exists")
-	}
+	w = <-ch
 	defer w.Stop()
 
 	keypair, caCertPool := w.KeypairAndCACertPool()
@@ -166,12 +157,7 @@ func TestKubernetesMount(t *testing.T) {
 	k8Setup(t, dir)
 
 	// the files exists now, expect the watcher to become ready.
-	var w *Watcher
-	select {
-	case w = <-ch:
-	case <-time.After(testReloadDelay):
-		t.Fatal("FutureWatcher should be ready once the TLS files exists")
-	}
+	w := <-ch
 	defer w.Stop()
 
 	expectedInitialCaCertPool := x509.NewCertPool()

--- a/pkg/crypto/certloader/watcher_test.go
+++ b/pkg/crypto/certloader/watcher_test.go
@@ -65,8 +65,20 @@ func TestRotation(t *testing.T) {
 	assert.Nil(t, err)
 	defer w.Stop()
 
+	prevKeypairGeneration, prevCaCertPoolGeneration := w.generations()
 	rotate(t, hubble, relay)
-	<-time.After(testReloadDelay)
+
+	// wait until both keypair and caCertPool have been reloaded
+	ticker := time.NewTicker(testReloadDelay)
+	defer ticker.Stop()
+	for range ticker.C {
+		keypairGeneration, caCertPoolGeneration := w.generations()
+		keypairUpdated := keypairGeneration > prevKeypairGeneration
+		caCertPoolUpdated := caCertPoolGeneration > prevCaCertPoolGeneration
+		if keypairUpdated && caCertPoolUpdated {
+			break
+		}
+	}
 
 	keypair, caCertPool := w.KeypairAndCACertPool()
 	assert.Equal(t, &expectedKeypair, keypair)
@@ -173,8 +185,20 @@ func TestKubernetesMount(t *testing.T) {
 	assert.Equal(t, &expectedInitialKeypair, keypair)
 	assert.Equal(t, expectedInitialCaCertPool.Subjects(), caCertPool.Subjects())
 
+	prevKeypairGeneration, prevCaCertPoolGeneration := w.generations()
 	k8sRotate(t, dir)
-	<-time.After(testReloadDelay)
+
+	// wait until both keypair and caCertPool have been reloaded
+	ticker := time.NewTicker(testReloadDelay)
+	defer ticker.Stop()
+	for range ticker.C {
+		keypairGeneration, caCertPoolGeneration := w.generations()
+		keypairUpdated := keypairGeneration > prevKeypairGeneration
+		caCertPoolUpdated := caCertPoolGeneration > prevCaCertPoolGeneration
+		if keypairUpdated && caCertPoolUpdated {
+			break
+		}
+	}
 
 	expectedRotatedCaCertPool := x509.NewCertPool()
 	if ok := expectedRotatedCaCertPool.AppendCertsFromPEM(rotatedHubbleServerCA); !ok {

